coordinator/src/main/scala/filodb.coordinator/NodeCoordinatorActor.scala

```diff
@@ -208,21 +245,29 @@ class NodeCoordinatorActor(metaStore: MetaStore,
       dsCoordNotify.clear()
       columnStore.reset()
 
-    case AddDatasetCoord(dataset, version, dsCoordRef) =>
+    case AddDatasetCoord(originator, dataset, version, dsCoordRef, reloadFlag) =>
       dsCoordinators((dataset, version)) = dsCoordRef
       context.watch(dsCoordRef)
+      if(reloadFlag){
+        self ! ReloadIngestionState(originator, dataset, version)
+      }
 
     case Terminated(childRef) =>
       dsCoordinators.find { case (key, ref) => ref == childRef }
-                    .foreach { case (key, _) =>
-                      logger.warn(s"Actor $childRef has terminated!  Ingestion for $key will stop.")
-                      dsCoordinators.remove(key)
+                    .foreach { case ((datasetRef,version), _) =>
+                      logger.warn(s"Actor $childRef has terminated!  Ingestion for ${(datasetRef,version)} will stop.")
+                      dsCoordinators.remove((datasetRef,version))
+                      // TODO: update ingestion state
+                      metaStore.updateIngestionState(actorPath,datasetRef,"Failed","Error during ingestion",version)
                     }
 
     case d @ DatasetCreateNotify(dataset, version, msg) =>
       logger.debug(s"$d")
       for { listener <- dsCoordNotify((dataset -> version)) } listener ! msg
       dsCoordNotify.remove((dataset -> version))
+
+    case ReloadDCA =>
```

Comment:
ReloadDCA -> this will be triggered from FiloSetup class, line no:137 and it's invoked during start up.
val resp = coordinatorActor ? ReloadDCA